processor/src/decorators/mod.rs

```diff
@@ -107,6 +110,22 @@ impl Process {
 
         Ok(())
     }
+
+    fn read_top_word(&self) -> Word {
+        [
+            self.stack.get(3),
+            self.stack.get(2),
+            self.stack.get(1),
+            self.stack.get(0),
+        ]
+    }
```

Comment:
I wonder if this method should be a part of the `Stack` struct. We could call it `get_top_word()`.